### PR TITLE
Make ContainerCluster abstract.

### PR DESCRIPTION
**Pull request by gjoranv**
This should not contain any functional changes, but I'd like to see it through factory before going further.

### CODE REVIEW

**Review by hmusum — APPROVED** [review]

File: config-model/src/main/java/com/yahoo/vespa/model/container/ContainerClusterImpl.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import com.yahoo.config.model.deploy.DeployState;
+import com.yahoo.config.model.producer.AbstractConfigProducer;
+
+/**
+ * The default container cluster implementation.
+ *
+ * @author gjoranv
+ */
+public final class ContainerClusterImpl
```
Comment (bjorncs) [inline config-model/src/main/java/com/yahoo/vespa/model/container/ContainerClusterImpl.java]: Consider renaming to `DefaultContainerCluster` if there will be multiple subclasses of `ContainerCluster`.

File: config-model/src/main/java/com/yahoo/vespa/model/container/ContainerClusterImpl.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
```
Comment (bjorncs) [inline config-model/src/main/java/com/yahoo/vespa/model/container/ContainerClusterImpl.java]: Old copyright header

File: config-model/src/main/java/com/yahoo/vespa/model/container/ContainerCluster.java [hunk 3]
```diff
@@ -604,8 +606,8 @@ public final void addPlatformBundle(Path bundlePath) {
     public void getConfig(BundlesConfig.Builder builder) {
         Stream.concat(applicationBundles.stream().map(FileReference::value),
                       platformBundles.stream()
-                                     .map(ContainerCluster::toFileReferenceString))
-                                     .forEach(builder::bundle);
+                              .map(ContainerCluster::toFileReferenceString))
+                .forEach(builder::bundle);
```
Comment (bjorncs) [inline config-model/src/main/java/com/yahoo/vespa/model/container/ContainerCluster.java]: Improves readability a lot 👍

**Review by bjorncs — APPROVED** [review]

**Comment by gjoranv** [comment]
Replaced by #8777

**Pull request closed without merging**